packages/manager/apps/pci-savings-plan/src/hooks/useFilteredConsumption.tsx

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useMemo, useState } from 'react';
+import { SavingsPlanFlavorConsumption } from '@/types/savingsPlanConsumption.type';
+import { getLastXMonths } from '@/utils/formatter/date';
+import { useSavingsPlanConsumption } from './useSavingsPlanConsumption';
+import { getBigestConsumption, isInstanceFlavor } from '@/utils/savingsPlan';
+
+export const useFilteredConsumption = (locale: string) => {
+  const lastXMonths = useMemo(() => getLastXMonths(locale), [locale]);
+  const [period, setPeriod] = useState(lastXMonths[0]);
+  const [flavor, setFlavor] = useState('');
+
+  const [monthName, year] = period.split(' ');
+  const date = new Date(Date.parse(`${monthName} 1, ${year}`));
+  const monthNumber = date.getMonth() + 1;
+
+  const {
+    data: consumption,
+    isLoading: isConsumptionLoading,
+  } = useSavingsPlanConsumption({
+    flavor: '',
+    year: year?.toString(),
+    month: monthNumber?.toString(),
+  });
+
+  const setFilterByPlan = (consumptionFlavor: SavingsPlanFlavorConsumption) => {
+    const defaultFlavor = consumptionFlavor.flavor;
+    setFlavor(defaultFlavor);
+  };
+
+  useEffect(() => {
+    if (consumption && consumption.flavors?.length > 0) {
+      setFilterByPlan(getBigestConsumption(consumption.flavors));
+    }
+  }, [isConsumptionLoading]);
```

Comment:
I don't understand the goal of this and why it only depend on isConsumptionLoading while it is not used